homeassistant/components/sensor/fail2ban.py

```diff
@@ -81,61 +82,58 @@ def state_attributes(self):
 
     @property
     def state(self):
-        """Return the most recently banned IP Address."""
-        return self.last_ban
+        """Return the number of currently banned IP Address."""
+        return self.number
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit_of_measurement of the device."""
+        return DEFAULT_UNITS
 
-    def update(self):
+    async def async_update(self):
         """Update the list of banned ips."""
-        if self.log_parser.timer():
-            self.log_parser.read_log(self.jail)
+        self.log_parser.read_log(self.jail)
```

Comment:
We're not allowed to do I/O directly in coroutines, ie in the event loop. Is there any reason why we should change to `async_update`?

Reply:
I obviously misunderstood your previous comment. I changed to async_update to remove the time logic from the sensor. Is there something I don't understand ?

Reply:
Just call the method `update`. This will work with our standard auto polling and also work with sync I/O.

Reply:
Doing so now.

Reply:
Make it a regular function, ie just `def`, not `async def`.

Reply:
It's done now